thorlcr/activities/diskread/thdiskreadslave.cpp

```diff
@@ -161,33 +155,28 @@ class CDiskRecordPartHandler : public CDiskPartHandlerBase
 protected:
     offset_t localRowOffset;
     CDiskReadSlaveActivityRecord &activity;
-    bool needsFileOffset;
+    bool needTransform = false;
+    RtlDynamicRowBuilder outBuilder;
 public:
     CDiskRecordPartHandler(CDiskReadSlaveActivityRecord &activity);
     ~CDiskRecordPartHandler();
-    inline void setNeedsFileOffset(bool tf) { needsFileOffset = tf; }
     virtual void getMetaInfo(ThorDataLinkMetaInfo &info, IPartDescriptor *partDesc);
     virtual void open();
     virtual void close(CRC32 &fileCRC);
     offset_t getLocalOffset()
     {
-        dbgassertex(needsFileOffset);
-        return localRowOffset;
+        return in->getLastRowOffset();
     }
     inline const void *nextRow()
     {
-        if (needsFileOffset)
-            localRowOffset = in->getOffset();       // shame this needed as a bit inefficient
         const void *ret = in->nextRow();
         if (ret)
             ++activity.diskProgress;
```

Comment:
Are these lines correct.  Who is responsible for updating diskProgress?  is it through queryProgress or this call?